### PR TITLE
PHOENIX-5618 IndexScrutinyTool fix for array type columns

### PR DESCRIPTION
**Pull request by gokceni**

### CODE REVIEW

File: phoenix-core/src/main/java/org/apache/phoenix/mapreduce/index/IndexScrutinyMapper.java [hunk 1]
```diff
@@ -372,7 +372,16 @@ private boolean compareValues(int startIndex, List<Object> targetValues,
...
             Object targetValue = targetValues.get(i);
             Object sourceValue = sourceValues.get(i);
-            if (targetValue != null && !targetValue.equals(sourceValue)) {
+            if (targetValue != null) {
+                if (sourceValue.getClass().isArray()) {
+                    if (compareArrayTypes(sourceValue, targetValue)) {
+                        continue;
+                    }
+                } else {
+                    if (targetValue.equals(sourceValue)) {
```
Comment (swaroopak) [inline phoenix-core/src/main/java/org/apache/phoenix/mapreduce/index/IndexScrutinyMapper.java]: what if nested arrays?

Comment (gokceni) [reply phoenix-core/src/main/java/org/apache/phoenix/mapreduce/index/IndexScrutinyMapper.java]: I don't think we support nested arrays in Phoenix as a column type

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/IndexScrutinyToolIT.java [hunk 2]
```diff
@@ -161,6 +162,54 @@ public IndexScrutinyToolIT(String dataTableDdl, String indexTableDdl) {
...
+            List<Job> completedJobs = runScrutiny(null, dataTableName, indexTableName);
+            Job job = completedJobs.get(0);
+            assertTrue(job.isSuccessful());
+            Counters counters = job.getCounters();
+            assertEquals(2, getCounterValue(counters, VALID_ROW_COUNT));
+            assertEquals(0, getCounterValue(counters, INVALID_ROW_COUNT));
+
+            // Now insert a different varbinary row
+//            conn.createStatement().execute(
+//                    String.format("ALTER INDEX %s ON %S disable", indexTableName, dataTableName));
```
Comment (gjacoby126) [inline phoenix-core/src/it/java/org/apache/phoenix/end2end/IndexScrutinyToolIT.java]: Please remove commented out code.

File: phoenix-core/src/main/java/org/apache/phoenix/mapreduce/index/IndexScrutinyMapper.java [hunk 3]
```diff
@@ -381,6 +390,27 @@ private boolean compareValues(int startIndex, List<Object> targetValues,
         return true;
     }
 
+    private boolean compareArrayTypes(Object sourceValue, Object targetValue) {
```
Comment (swaroopak) [inline phoenix-core/src/main/java/org/apache/phoenix/mapreduce/index/IndexScrutinyMapper.java]: I think the switch-case will make this block more readable.

**Pull request merged**